plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
Also, having to do this sort of manual permissions is tedious - I don't think we've got a good handle on how to safely grant this permission to a set of pods yet.